File: src/main/java/io/github/oleksivio/telegram/bot/api/model/method/message/EditOwnMessageCaption.java
package io.github.oleksivio.telegram.bot.api.model.method.message;

import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;
import io.github.oleksivio.telegram.bot.core.model.method.message.EditMessageCaption;
import io.github.oleksivio.telegram.bot.core.model.responses.CommonResponse;
import io.github.oleksivio.telegram.bot.core.model.responses.MessageResponse;
import io.github.oleksivio.telegram.bot.api.model.objects.std.Message;

public class EditOwnMessageCaption extends EditMessageCaption<Message> {

    public EditOwnMessageCaption(ActionNetworker actionNetworker) {
        super(actionNetworker);
    }

    @Override
    protected Class<? extends CommonResponse<Message>> getResultWrapperClass() {
        return MessageResponse.class;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/message/EditOwnMessageCaption.kt
package io.github.oleksivio.telegram.bot.api.model.method.message

import io.github.oleksivio.telegram.bot.api.model.objects.std.Message
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker
import io.github.oleksivio.telegram.bot.core.model.MessageResponse
import io.github.oleksivio.telegram.bot.core.model.method.message.EditMessageCaption

class EditOwnMessageCaption(actionNetworker: ActionNetworker) : EditMessageCaption<Message>(actionNetworker) {

    override val resultWrapperClass = MessageResponse::class
}
